### PR TITLE
frontend: error notification component

### LINKED ISSUE #119: frontend error message handling alert

### PR DESCRIPTION
**Pull request by bwdmonkey**
:tickets: **Ticket(s)**: Closes #119

---

## :construction_worker: Changes

Introduces new `Notification` module. Overwrote usage of `id`/`type` with `name` to allow autocomplete.

## :flashlight: Testing Instructions

Try submitting on `/login` or `/signup` without filling in the fields.

### CODE REVIEW

**Comment by leesw98** [comment]
screen shot 2019-02-20 at 5 10 55 pm

**Review by zhulia96 — DISMISSED** [review]

File: frontend/src/components/Login.js [hunk 1]
```diff
@@ -29,54 +41,39 @@ class Login extends Component {
...
       if (resp.status === 200) {
         const { router: { history } } = this.context;
         history.push('/');
       } else {
-        this.setState({ message: { messageType: 'error', content: ' Incorrect Credentials.' } });
+        this.setState({
+          notification: {
+            type: 'error',
+            message: 'Incorrect Credentials.',
+            showNotification: true,
```
Comment (bobheadxi) [inline frontend/src/components/Login.js]: hmm, is there a particular usecase for `showNotification`? would setting `notification: null` to disable the notification do the trick?

File: frontend/src/components/Login.js [hunk 2]
```diff
@@ -29,54 +41,39 @@ class Login extends Component {
...
-    }
-  }
-
   render() {
+    const { notification } = this.state;
     return (
       <div className="flex-al-center">
         <div className="title margin-title">Sign In</div>
-        { this.generateMessage() }
+        <Notification {...notification} />
```
Comment (bobheadxi) [inline frontend/src/components/Login.js]: very clean 😍

File: frontend/src/components/Login.js (hunk 1 above)
Comment (leesw98) [reply frontend/src/components/Login.js]: O. I thought destructuring would break with null value but I just tested and nope. Will change 👍

Comment (bobheadxi) [reply frontend/src/components/Login.js]: You can also destructure with a default:

```js
const { notification = { defaults... } } = this.state
```

**Review by bobheadxi — CHANGES_REQUESTED** [review]
😢 https://travis-ci.com/ubclaunchpad/pinpoint/jobs/179382146#L507

**Review by bobheadxi — APPROVED** [review]
looks good to me! great stuff! 🚀

**Pull request merged**